### PR TITLE
Add codeql scanning

### PR DESCRIPTION
**Pull request by diracdeltas**

### CODE REVIEW

**Review by remusao — APPROVED** [review]
Thank you @diracdeltas

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
```
Comment (remusao) [inline .github/workflows/codeql-analysis.yml:48]: > The workflow is not valid. .github/workflows/codeql-analysis.yml (Line: 35, Col: 19): Matrix vector 'language' does not contain any values

**Review by remusao — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (diracdeltas) [reply .github/workflows/codeql-analysis.yml:48]: weird, this file was autogenerated by github. i'll take a look.

**Comment by diracdeltas** [comment]
i think this is broken because it was set up before your first commit was merged. going to close then redo.

**Pull request closed without merging**